### PR TITLE
Adds sessions/create route & session, speaker form

### LINKED ISSUE #757: Implement speakers & sessions form using custom-form API
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [x] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-frontend

- Implement speakers & sessions form using custom-form API
- Implement session creation for events/<id> route

### PR DESCRIPTION
**Pull request by geekyd**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:
- Adds sessions/create route & session, speaker form

Fixes #757, #359

### CODE REVIEW

**Comment by niranjan94** [comment]
Deployment link: https://pr-770-open-event-frontend-fossasia.surge.sh

**Comment by niranjan94** [comment]
Deployment link: https://pr-770-open-event-frontend-fossasia.surge.sh

**Comment by niranjan94** [comment]
Deployment link: https://pr-770-open-event-frontend-fossasia.surge.sh

File: app/serializers/event.js [hunk 1]
```diff
@@ -5,10 +5,11 @@ export default ApplicationSerializer.extend(CustomPrimaryKeyMixin, {
...
   attrs      : {
-    type      : 'event-type',
-    topic     : 'event-topic',
-    subTopic  : 'event-sub-topic',
-    copyright : 'event-copyright'
+    type        : 'event-type',
+    topic       : 'event-topic',
+    subTopic    : 'event-sub-topic',
+    copyright   : 'event-copyright',
+    customForms : 'custom-form'
```
Comment (CosmicCoder96) [inline app/serializers/event.js]: @geekyd Why is this needed?

File: app/mixins/custom-form.js [hunk 2]
```diff
@@ -16,96 +16,96 @@ export default Mixin.create(MutableArray, {
...
+        type            : 'select',
         isRequired      : false,
         isIncluded      : false,
         event           : parent
       }),
       this.store.createRecord('custom-form', {
-        fieldIdentifier : 'Slides',
+        fieldIdentifier : 'slidesUrl',
         form            : 'session',
         type            : 'text',
```
Comment (niranjan94) [inline app/mixins/custom-form.js]: Type file.

File: app/mixins/custom-form.js [hunk 3]
```diff
@@ -16,96 +16,96 @@ export default Mixin.create(MutableArray, {
...
         isRequired      : false,
         isIncluded      : true,
         event           : parent
       }),
       this.store.createRecord('custom-form', {
-        fieldIdentifier : 'Video',
+        fieldIdentifier : 'videoUrl',
         form            : 'session',
         type            : 'text',
```
Comment (niranjan94) [inline app/mixins/custom-form.js]: Type file.

File: app/mixins/custom-form.js [hunk 4]
```diff
@@ -16,96 +16,96 @@ export default Mixin.create(MutableArray, {
...
         isRequired      : false,
         isIncluded      : false,
         event           : parent
       }),
       this.store.createRecord('custom-form', {
-        fieldIdentifier : 'Audio',
+        fieldIdentifier : 'audioUrl',
         form            : 'session',
         type            : 'text',
```
Comment (niranjan94) [inline app/mixins/custom-form.js]: Type file.

File: app/serializers/event.js (hunk 1 above)
Comment (geekyd) [inline app/serializers/event.js]: @CosmicCoder96 Thanks for pointing it out :).

**Comment by niranjan94** [comment]
Deployment link: https://pr-770-open-event-frontend-fossasia.surge.sh

**Comment by geekyd** [comment]
@CosmicCoder96 @niranjan94 Made the changes please review. Thanks

File: app/templates/components/forms/session-speaker-form.hbs [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    {{/if}}
+    {{#each allFields.speaker as |field|}}
+      {{#if field.isIncluded}}
+        <div class="field">
+          <label class="{{if field.isRequired 'required'}}" for="name">{{field.name}}</label>
+          {{#if (or (eq field.type 'text') (eq field.type 'email'))}}
+            {{#if field.isLongText}}
+              {{widgets/forms/rich-text-editor textareaId=(if field.isRequired (concat 'speaker_' field.fieldIdentifier '_required'))}}
+            {{else}}
+              {{input type=field.type id=(if field.isRequired (concat 'speaker_' field.fieldIdentifier '_required'))}}
```
Comment (niranjan94) [inline app/templates/components/forms/session-speaker-form.hbs]: We need a component similar to `widgets/forms/image-upload` which would upload the file to the API (with progress). But I think that could be handled in a separate issue.

File: app/mixins/custom-form.js [hunk 6]
```diff
@@ -16,96 +16,96 @@ export default Mixin.create(MutableArray, {
...
         isRequired      : false,
         isIncluded      : true,
         event           : parent
       }),
       this.store.createRecord('custom-form', {
-        fieldIdentifier : 'Video',
+        fieldIdentifier : 'videoUrl',
         form            : 'session',
-        type            : 'text',
+        type            : 'file',
```
Comment (niranjan94) [inline app/mixins/custom-form.js]: Should be file or url. We need to give an option for users to choose.

Comment (geekyd) [inline app/mixins/custom-form.js]: @niranjan94 The server only supports `file` that's why I have used it instead of file or url, we can take care of this in the component for file upload itself.

File: app/templates/components/forms/session-speaker-form.hbs (hunk 5 above)
Comment (geekyd) [inline app/templates/components/forms/session-speaker-form.hbs]: @niranjan94 I will add this component while implementing session creation via API.

Comment (geekyd) [inline app/templates/components/forms/session-speaker-form.hbs]: Will be taken care of in https://github.com/fossasia/open-event-frontend/issues/792

File: app/mixins/custom-form.js (hunk 6 above)
Comment (niranjan94) [inline app/mixins/custom-form.js]: The server expects a URL to the file. Thats it. So that URL could be directly entered by the user itself.

File: app/templates/components/forms/session-speaker-form.hbs (hunk 5 above)
Comment (niranjan94) [inline app/templates/components/forms/session-speaker-form.hbs]: Cool

File: app/mixins/custom-form.js (hunk 6 above)
Comment (geekyd) [inline app/mixins/custom-form.js]: @niranjan94 Okay, If the user uploads a file the server will return a URL. What I meant was the custom-form API expects one of these values as type image, text, select, email, file. So thats why we have file here. 

We will be allowing the user to enter the URL or upload a file on the FE.

Comment (niranjan94) [inline app/mixins/custom-form.js]: Okay cool

**Comment by niranjan94** [comment]
Deployment link: https://pr-770-open-event-frontend-fossasia.surge.sh

**Comment by geekyd** [comment]
@niranjan94 @CosmicCoder96 Please review it again. Thanks

**Comment by niranjan94** [comment]
Deployment link: https://pr-770-open-event-frontend-fossasia.surge.sh
